### PR TITLE
Shows feedback blocks instantly

### PR DESCRIPTION
**Pull request by Hamms**
Previously, when showing blocks in the feedback, we would insert the
iframe containing the blocks into the dom, then remove it, then wait for
the "show hint" button to be pressed, then re-add it to the dom.

Since iframes don't load unless they're in the dom, this could cause a
noticable delay between the button being pressed and the blocks
appearing, particularly on slower connections.

Instead, after initially inserting the iframe into the dom, we now
simply hide it, and reveal it when the button is pressed. Thus, the
blocks can load in the background before the user presses the button.

| Before | After |
| --- | --- |
| before | after |

Note that here I'm on a relatively fast connection and computer; the delay on slower connections can be quite a bit more significant.

### CODE REVIEW

**Comment by bcjordan** [comment]
Re: testing, might be worth running `feedback.feature` with chromedriver pre-merge

File: apps/src/feedbackBlocks.js [hunk 1]
```diff
@@ -85,6 +86,14 @@ FeedbackBlocks.prototype.show = function() {
   }
 };
 
+FeedbackBlocks.prototype.hideDiv = function() {
+  this.div.className += " hidden";
+};
+
+FeedbackBlocks.prototype.revealDiv = function() {
+  this.div.className = this.div.className.replace(/\hidden\b/,'');
```
Comment (bcjordan) [inline apps/src/feedbackBlocks.js]: What do the `\h` and `\b` do in this?

If it's safe to use jQuery in this context it might be worth using `$(this.div).removeClass('hidden');`

Alternatively there's [blockly's implementation](https://github.com/code-dot-org/code-dot-org/blob/c692117679fe0c8aa9284b4f7459f1e5b10700c2/blockly-core/core/utils/utils.js#L66-L66) but might be overkill since we're not dealing with an svg el't here. Or [YMNNJq's version](http://youmightnotneedjquery.com/#remove_class). If using something like that or if it's a known idiom, could be worth commenting w/ link to source

File: apps/src/feedback.js [hunk 2]
```diff
@@ -232,15 +232,9 @@ FeedbackUtils.prototype.displayFeedback = function(options, requiredBlocks,
       // feedback block
       var genericFeedback = this.getFeedbackMessage_({message: msg.tryBlocksBelowFeedback()});
 
-      // If there are feedback blocks, temporarily remove them.
-      // Get pointers to the parent and next sibling so we can re-insert
-      // the feedback blocks into the correct location if needed.
-      var feedbackBlocksParent = null;
-      var feedbackBlocksNextSib = null;
```
Comment (bcjordan) [inline apps/src/feedback.js:239]: Love the simplification!

**Comment by bcjordan** [comment]
LGTM overall! Just one Q above & one below.

Re: manual testing, could be worth manual testing IE9 (or even 10 if pain to test against local, then 9 once hits staging), Firefox and maybe Safari since we sometimes see show/hide issues there, & resized/hidden iframes w/ SVG in them sound ripe for edge cases (but no specific reason to believe would be an issue) :)

File: dashboard/app/assets/stylesheets/application.css.scss [hunk 3]
```diff
@@ -1950,6 +1950,11 @@ $modal-no-icon-gap: 22px;
   }
 }
 
+#feedbackBlocksContainer.hidden {
```
Comment (bcjordan) [inline dashboard/app/assets/stylesheets/application.css.scss]: Know what, it might be worth naming this something other than `hidden` (maybe `.hidden-iframe`?), even though it's scoped, just in case something gets called `.hidden` on a level page down the road that adds another property / hides in an unexpected way. E.g., netsim has its own .hidden that does `display: none` (but fortunately we don't use blocks in that app)

File: apps/src/feedbackBlocks.js (hunk 1 above)
Comment (Hamms) [inline apps/src/feedbackBlocks.js]: hah, oh gosh, that's a typo. Supposed to be `/\bhidden\b/`; `\b` enforces a word boundary. so it wouldn't select "hiddenSomethingElse".

I was debating exactly that; it's probably safe to use jQuery in this context, but I wanted to avoid adding a dependency to this class if I didn't have to.

This regex is pretty functionally similar to YMNNJq's version, and is I believe a common approach. Would a comment just explaining the regex be sufficient?

**Comment by Hamms** [comment]
I did run feedback.feature on a local SauceConnect against our full grid, but I'll definitely double-check manually on the IEs.

File: apps/src/feedbackBlocks.js (hunk 1 above)
Comment (bcjordan) [inline apps/src/feedbackBlocks.js]: Yup, sounds good!

**Comment by Hamms** [comment]
Manually tested on FF, IE9, and Safari!
